### PR TITLE
Week3

### PR DESCRIPTION
**Pull request by amagsid**
3/4 exercises of homework done due to losing access to MySQL while trying to solve an error in ex.4

### CODE REVIEW

File: Week3/Homework/3.SQL-injection/SQL-injection.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+"use strict";
+
+//value passed as name and can take advantage of SQL injection: Mars or 1=1 ; SHOW TABLES
```
Comment (GinaSt) [inline Week3/Homework/3.SQL-injection/SQL-injection.js:3]: Not exactly! You need to pay attention to the quotes! `Mars' or' 1=1`

File: Week3/Homework/4.Mongo-CRUD/mongo-CRUD.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+  });
+  try {
+    await client.connect();
+
+    const db = client.db("World");
+    const cityCollection = await db.collection("City");
+
+    new city();
```
Comment (GinaSt) [inline Week3/Homework/4.Mongo-CRUD/mongo-CRUD.js:19]: is this used somewhere?

**Review by GinaSt — COMMENTED** [review]
Hello! 
You are almost done!
I have left various comments - maybe you will get inspired for some improvements.

File: Week3/Homework/1.Normalization/Normalization.mkd [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# What columns violate 1NF?
+
+    Columns that violate the 1NF are: dinner_date, food_code, food_description
```
[the remark this answers is not in the document]
Comment (amagsid) [reply Week3/Homework/1.Normalization/Normalization.mkd:3]: I fixed that and elaborated why.

File: Week3/Homework/1.Normalization/Normalization.mkd [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+## What entities do you recognize that could be extracted?
+
+    member, dinner, venue, food
+
+## Name all the tables and columns that would make a 3NF compliant solutionm
+
```
[the remark this answers is not in the document]
Comment (amagsid) [reply Week3/Homework/1.Normalization/Normalization.mkd:11]: Thanks for clarifying. Also fixed. Hope I got it right this time.

File: Week3/Homework/3.SQL-injection/SQL-injection.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+//value passed as code and can take advantage of SQL injection: INJCT or 1=1; DROP DATABASE World
+
+const getPopulation = (Country, name, code, cb) => {
+  conn.query(
+    `SELECT Population FROM ${Country} WHERE Name =` +
```
[the remark this answers is not in the document]
Comment (amagsid) [reply Week3/Homework/3.SQL-injection/SQL-injection.js:8]: I did! I also used it in some other exercise the week before.

File: Week3/Homework/4.Mongo-CRUD/mongo-CRUD.js (hunk 2 above)
Comment (amagsid) [reply Week3/Homework/4.Mongo-CRUD/mongo-CRUD.js]: After a double check, no. Fixed.

**Comment by amagsid** [comment]
@GinaSt Thanks for taking the time to review and give feedback!

**Pull request closed without merging**